Avoid overlinking abseil in libs using grpc

grpc exports a pkg-config file that instructs all users of grpc to link
against some libraries from libabseil when linking against grpc. Because
abseil often changes its soversion, this requires many ports that use
grpc to be revbumped whenever abseil changes, even though they only use
symbols from grpc.

Avoid this by using the Requires.private field in the grpc pkg-config
file as documented in [1]. See also [2], which explains the difference
in linking.

[1]: https://people.freedesktop.org/~dbn/pkg-config-guide.html#concepts
[2]: https://stackoverflow.com/a/61674208

Upstream-Status: Pending
--- ./cmake/pkg-config-template.pc.in.orig	2025-05-08 11:14:03
+++ ./cmake/pkg-config-template.pc.in	2025-05-08 11:14:17
@@ -7,7 +7,6 @@
 Description: @PC_DESCRIPTION@
 Version: @PC_VERSION@
 Cflags: -I${includedir}
-Requires: @PC_REQUIRES@
 Requires.private: @PC_REQUIRES_PRIVATE@
 Libs: -L${libdir} @PC_LIB@
 Libs.private: @PC_LIBS_PRIVATE@
